sunpy/net/vso/vso.py

```diff
@@ -710,17 +710,52 @@ def create_getdatarequest(self, maps, methods, info=None):
         fileids and methods, """
         if info is None:
             info = {}
-
+        if maps.get('JSOC',0):
+            return self._separate_hmi(maps, methods, info)
+        else:
+            return self.make(
+                'VSOGetDataRequest',
+                request__method__methodtype=methods,
+                request__info=info,
+                request__datacontainer__datarequestitem=[
+                    self.make('DataRequestItem', provider=k, fileiditem__fileid=[v])
+                    for k, v in iteritems(maps)
+                ]
+            )
+    def _separate_hmi(self, maps_dict, methods, info):
+        """
+        This function parses the separate hmi data series from the
+        maps_dict(jsoc) list, pulls any hmi data it finds into separate lists
+        and deletes the hmi data from the original jsoc list.  It then continues
+        as normal to construct the soap message.
+        """
+        jsoc_list=maps_dict.get('JSOC',0)
+        if jsoc_list:
+            listdict={}
+            s_match=re.compile('^.*?:')
+            for fidnum, fid in enumerate(jsoc_list):
+                if re.search('^hmi',fid):
+                    #hmi data match found, extract into new list
+                    series=s_match.findall(fid)[0]
+                    if listdict.get(series,0):
+                        listdict.get(series).append(fid)
+                    else:
+                        listdict[series]=[fid]
+                    #delete entry from jsoc_list
+                    del jsoc_list[fidnum]
+            new_dict=maps_dict.copy()
+            new_dict.update(listdict)
+        #new dict defined
+        for keyn, key in enumerate(new_dict):
```

Comment:
What happens if above the `if` branch is not done? `new_dict` doesn't exist... and it will break here.